### PR TITLE
enh(cpp) Highlight all function dispatches

### PR DESCRIPTION
**Pull request by joshgoebel**
Work towards resolving #2973.

Related: #2500.

### Changes

- Highlights all function dispatch as `function.dispatch`, which aliases
to `built_in` for now

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md`
- [x] Added myself to `AUTHORS.txt`, under Contributors

### CODE REVIEW

**Comment by allejo** [comment]
Should `_` be a keyword for an "only used for relevancy" list (or was this just a temporary name)? And for the Arduino grammar, was your split of `built_in` keywords arbitrary was there some meaning to how you split them?

**Comment by joshgoebel** [comment]
> Should _ be a keyword for an "only used for relevancy" list (or was this just a temporary name)? 

It's a convention, though I also considered `_*` as in `_builtins`, `_something`, `_whatever` to allow for better naming.

> And for the Arduino grammar, was your split of built_in keywords arbitrary was there some meaning to how you split them?

It's split cleanly between built-in classes (still highlighted as keywords) and built-in functions (highlighted via the `blahBlah(` rule now.

**Comment by joshgoebel** [comment]
Ping.

**Comment by allejo** [comment]
> It's a convention now (hence the patch to avoid emitting HTML for them), though I also considered `_*` as in `_builtins`, `_something`, `_whatever` to allow for better naming.

Yea I noticed the patch in the core changes. I'm not convinced that `_` is the best name for this. I'd lean towards using something like `_relevancy`. Then in the future, `_*` could be used for future internal or special behavior stuff.

**Comment by joshgoebel** [comment]
>  Then in the future, _* could be used for future internal or special behavior stuff.

I like free naming too much so I codified `_*` and added it to docs. I do think it's an improvement from the magic `_` key.  We always have `__`  or `$` or all sort of options for adding future functionality including passing an object:

```js
keywords: {
  delicious: { extraSpecialHighlighting: true, list: "for if def when" }
}
```

I don't want to worry too much about saving the namespace for the future.

File: src/languages/cpp.js [hunk 1]
```diff
@@ -120,19 +236,27 @@ export default function(hljs) {
...
-      'auto_ptr deque list queue stack vector map set pair bitset multiset multimap unordered_set ' +
-      'unordered_map unordered_multiset unordered_multimap priority_queue make_pair array shared_ptr abort terminate abs acos ' +
-      'asin atan2 atan calloc ceil cosh cos exit exp fabs floor fmod fprintf fputs free frexp ' +
-      'fscanf future isalnum isalpha iscntrl isdigit isgraph islower isprint ispunct isspace isupper ' +
-      'isxdigit tolower toupper labs ldexp log10 log malloc realloc memchr memcmp memcpy memset modf pow ' +
-      'printf putchar puts scanf sinh sin snprintf sprintf sqrt sscanf strcat strchr strcmp ' +
-      'strcpy strcspn strlen strncat strncmp strncpy strpbrk strrchr strspn strstr tanh tan ' +
-      'vfprintf vprintf vsprintf endl initializer_list unique_ptr _Bool complex _Complex imaginary _Imaginary',
+    built_in: '_Bool _Complex _Imaginary',
+    _relevance_hints: COMMON_CPP_HINTS,
```
Comment (allejo) [inline src/languages/cpp.js:240]:
```suggestion
    _relevance_only: COMMON_CPP_HINTS,
```

I know it doesn't matter but just to match the docs

**Review by allejo — APPROVED** [review]

File: src/languages/cpp.js (hunk 1 above)
Comment (joshgoebel) [reply src/languages/cpp.js:240]: I prefere hints (of the two), but it's not a magic phrase - they don't have to match.

**Pull request merged**